cmake/Modules/BuildCommon.cmake

```diff
@@ -469,18 +470,34 @@ function(add_katana_sphinx_target target_name)
   endif ()
 
   get_target_property(PYTHON_ENV_SCRIPT ${target_name} PYTHON_ENV_SCRIPT)
-
-  add_custom_target(
+  if (NOT BUILD_EXTERNAL_DOCS)
+    set(sphinx_build_flags "-W -t internal")
+    add_custom_target(
       ${target_name}_sphinx_docs
       COMMAND ${CMAKE_COMMAND} -E rm -rf ${CMAKE_BINARY_DIR}/docs/${target_name}
       COMMAND env KATANA_DOXYGEN_PATH="${CMAKE_BINARY_DIR}/docs/doxygen" ${PYTHON_ENV_SCRIPT} sphinx-build
         -W
         -b html
+        -t internal
```

Comment:
Why didn't something like this work?

```cmake
set(tag_option "-t external")
if(BUILD_DOCS STREQUAL "internal")
  set(tag_option "-t internal")
endif()

add_custom_target(... ${tag_option})
```

Reply:
not sure. I tried that exact thing but for some reason it wasn't getting picked up

Reply:
added comment